### PR TITLE
add: ticket categories functions

### LINKED ISSUE #18046: Missing tags and categories in tickets
# Feature Request
Missing tags and categories in tickets

### PR DESCRIPTION
**Pull request by evarisk-theo**
Fix #18046 

PR 2/3 add: functions of ticket categories

### CODE REVIEW

File: htdocs/core/class/html.form.class.php [hunk 1]
```diff
@@ -6582,6 +6582,218 @@ public function select_duration($prefix, $iSecond = '', $disabled = 0, $typehour
 		return;
 	}
 
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
```
Comment (eldy) [inline htdocs/core/class/html.form.class.php:6585]: For new inserted method, we must avoid this: Just rename your new method
select_ticket into
selectTicket
so you can remove the phpcs:disable and phpcs:enable tag

Comment (evarisk-theo) [reply htdocs/core/class/html.form.class.php]: ok thank you, it's fixed

File: htdocs/core/class/html.form.class.php [hunk 2]
```diff
@@ -6582,6 +6581,214 @@ public function select_duration($prefix, $iSecond = '', $disabled = 0, $typehour
...
+	 *	@param      int		$limit              Limit on number of returned lines
+	 * 	@param      string	$filterkey          Filter on product
+	 *	@param		int		$status             Ticket status
+	 *  @param      int		$outputmode         0=HTML select string, 1=Array
+	 *  @param		string	$showempty		    '' to not show empty line. Translation key to show an empty line. '1' show empty line with no text.
+	 * 	@param		int		$forcecombo		    Force to use combo box
+	 *  @param      string  $morecss            Add more css on select
+	 *  @return     array    				    Array of keys for json
+	 */
+	public function select_tickets_list($selected = '', $htmlname = 'ticketid', $filtertype = '', $limit = 20, $filterkey = '', $status = 1, $outputmode = 0, $showempty = '1', $forcecombo = 0, $morecss = '')
```
Comment (eldy) [inline htdocs/core/class/html.form.class.php:6662]: Same for this new method. Use camelcase naming:
selectTicketList

Comment (evarisk-theo) [reply htdocs/core/class/html.form.class.php]: it's done ! :)

**Pull request merged**